openassessment/xblock/openassessmentblock.py

```diff
@@ -246,6 +247,27 @@ class OpenAssessmentBlock(MessageMixin,
         help="Indicates whether or not there are peers to grade."
     )
 
+    label_list = List(
+        default = DEFAULT_LABEL_LIST,
+        scope = Scope.content,
+        help="List of strings relavent to the content."
+    )
+
+    @property
+    def labels(self):
+        return ",".join(self.label_list)
+
+    @labels.setter
+    def labels(self, value):
+        """
+        Removes duplicates and stores label list.
+        """
+        if value is None or value.strip() == "":
+            self.label_list = []
+        else:
+            label_list = [label.strip() for label in value.split(",")]
+            self.label_list = list(OrderedDict.fromkeys(label_list))
```

Comment:
We don't have a clear description of what a label can and cannot be at this point. There are many things to consider, for example, special characters and numbers. Since the string is Unicode, it can have characters from other languages too. So, I've left things for when we have more details of the requirement.

Reply:
Added label formatting rules and validation.